zebra-rpc/src/methods/get_block_template_rpcs/types/long_poll.rs

```diff
@@ -0,0 +1,276 @@
+    /// was generated. This might be different from the number of transactions in the template,
+    /// due to ZIP-317.
+    ///
+    /// If the number of mempool transactions changes, a new template might be provided.
+    /// Old work is still valid.
+    ///
+    /// The number of transactions is limited by the mempool DoS limit.
+    ///
+    /// Using the number of transactions makes mempool checksum attacks much harder.
+    /// It also helps with debugging, and reduces the probability of a missed mempool change.
+    pub mempool_transaction_count: u32,
+
+    /// A checksum of the effecting hashes of the transactions in the mempool,
+    /// when the template containing this long poll ID was generated.
+    /// We ignore changes to authorizing data.
+    ///
+    /// This might be different from the transactions in the template, due to ZIP-317.
+    ///
+    /// If the content of the mempool changes, a new template might be provided.
+    /// Old work is still valid.
+    ///
+    /// This checksum is not cryptographically secure.
+    ///
+    /// It's ok to do a probabilistic check here,
+    /// so we choose a 1 in 2^32 chance of missing a transaction change.
+    ///
+    /// # Security
+    ///
+    /// Attackers could use dust transactions to keep the checksum at the same value.
+    /// But this would likely change the number of transactions in the mempool.
+    ///
+    /// If an attacker could also keep the number of transactions constant,
+    /// a new template will be generated when the tip hash changes, or the max time is reached.
+    pub mempool_transaction_content_checksum: u32,
+}
+
+impl From<LongPollInput> for LongPollId {
+    /// Lossy conversion from LongPollInput to LongPollId.
+    fn from(input: LongPollInput) -> Self {
+        let mut tip_hash_checksum = 0;
+        update_checksum(&mut tip_hash_checksum, input.tip_hash.0);
+
+        let mut mempool_transaction_content_checksum: u32 = 0;
+        for tx_mined_id in input.mempool_transaction_mined_ids.iter() {
+            update_checksum(&mut mempool_transaction_content_checksum, tx_mined_id.0);
+        }
+
+        Self {
+            tip_height: input.tip_height.0,
+
+            tip_hash_checksum,
+
+            // It's ok to do wrapping conversions here,
+            // because long polling checks are probabilistic.
+            max_timestamp: input.max_time.timestamp() as u32,
+
+            mempool_transaction_count: input.mempool_transaction_mined_ids.len() as u32,
+
+            mempool_transaction_content_checksum,
+        }
+    }
+}
+
+/// Update `checksum` from `item`, so changes in `item` are likely to also change `checksum`.
+///
+/// This checksum is not cryptographically secure.
+fn update_checksum(checksum: &mut u32, item: [u8; 32]) {
+    for chunk in item.chunks(4) {
+        let chunk = chunk.try_into().expect("chunk is u32 size");
+        let chunk = u32::from_le_bytes(chunk);
```

Comment:
Why call `from_le_bytes` here?

Reply:
We have to convert using some method, and it doesn't really matter which one.
I added a comment in 9cea1242ae.